lib/active_fedora/file.rb

```diff
@@ -176,6 +170,12 @@ def prefix(path)
 
     private
 
+      def create_or_update(*options)
+        super.tap do
+          metadata.save if metadata.changed?
```

Comment:
It's possible, but that should be in a different PR.  This PR is about moving the logic to be called when `save` or `save!` are called.